internal/resources/app_endpoint_activation_status.go

```diff
@@ -0,0 +1,368 @@
+		online          = plan.Online.ValueBool()
+	)
+
+	if err := r.manageAppEndpointActivation(ctx, online, organizationId, projectId, clusterId, appServiceId, appEndpointName); err != nil {
+		resp.Diagnostics.AddError(
+			"App Endpoint activation failed",
+			err.Error(),
+		)
+		return
+	}
+
+	refreshedState, err := r.waitForAppEndpointStatus(ctx, organizationId, projectId, clusterId, appServiceId, appEndpointName, online)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error Reading Capella App Endpoint Activation Status",
+			api.ParseError(err),
+		)
+		return
+	}
+
+	diags = resp.State.Set(ctx, refreshedState)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Delete is a no-op as the activation resource models an action.
+// so deleting will only remove it from the state file.
+func (r *AppEndpointActivationStatus) Delete(_ context.Context, _ resource.DeleteRequest, _ *resource.DeleteResponse) {
+	// Couchbase Capella's v4 does not support destroying an activation resource.
+	// The POST and DELETE endpoints are used to switch the app endpoint online and offline respectively.
+	// https://docs.couchbase.com/cloud/management-api-reference/index.html#tag/App-Endpoints/operation/postAppEndpointActivationStatus
+	// https://docs.couchbase.com/cloud/management-api-reference/index.html#tag/App-Endpoints/operation/deleteAppEndpointActivationStatus
+}
+
+// retrieveAppEndpointActivation reads the app endpoint and maps its state to the Online flag.
+func (r *AppEndpointActivationStatus) retrieveAppEndpointActivation(ctx context.Context, organizationId, projectId, clusterId, appServiceId, appEndpointName string) (*app_service_api.GetAppEndpointStateResp, error) {
+	url := fmt.Sprintf("%s/v4/organizations/%s/projects/%s/clusters/%s/appservices/%s/appEndpoints/%s", r.HostURL, organizationId, projectId, clusterId, appServiceId, appEndpointName)
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodGet, SuccessStatus: http.StatusOK}
+	response, err := r.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		nil,
+		r.Token,
+		nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	var getResp app_service_api.GetAppEndpointStateResp
+	if err := json.Unmarshal(response.Body, &getResp); err != nil {
+		return nil, err
+	}
+
+	return &getResp, nil
+}
+
+// waitForAppEndpointStatus monitors the status of an App Endpoint online/offline request.
+// It periodically fetches the App Endpoint status using the `retrieveAppEndpointActivation`
+// function and waits until the App Endpoint reaches the desired state or until a specified timeout is reached.
+// The function returns an error if the operation times out or encounters an error during status retrieval.
+func (r *AppEndpointActivationStatus) waitForAppEndpointStatus(ctx context.Context, organizationId, projectId, clusterId, appServiceId, appEndpointName string, online bool) (*providerschema.AppEndpointActivationStatus, error) {
```

Comment:
in this case no need to return `providerschema.AppEndpointActivationStatus`.  there are no computed attributes.